src/gpu/nvidia/cudnn_inner_product.cpp

```diff
@@ -72,12 +69,13 @@ status_t cudnn_inner_product_fwd_t::execute(const exec_ctx_t &ctx) const {
 
             std::vector<void *> args;
 
-            args.push_back(sc.memory<void *>(ih, src_acc));
-            args.push_back(sc.memory<void *>(ih, wei_acc));
+            args.push_back(arg_src.get_native_pointer(ih, sc));
+            args.push_back(arg_wei.get_native_pointer(ih, sc));
             args.push_back(
-                    ((pd()->with_bias()) ? sc.memory<void *>(ih, *bias_acc)
+                    ((pd()->with_bias()) ? arg_bias.get_native_pointer(ih, sc)
```

Comment:
`get_native_pointer(...)` returns `nullptr` when it's empty so we can just do `args.push_back(arg_bias.get_native_pointer(ih))`.

Reply:
done